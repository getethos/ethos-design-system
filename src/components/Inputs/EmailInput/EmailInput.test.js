import React from 'react'
import { EmailInput } from './index.js'
import renderer from 'react-test-renderer'

// TODO: make a helper
// https://github.com/text-mask/text-mask/issues/427
jest.mock('react-text-mask', () => (props) => <input type="email" {...props} />)

const formChangeHandlerStub = jest.fn()

describe('EmailInput', () => {
  it('default rendering', () => {
    const tree = renderer
      .create(
        <EmailInput
          name="the-email-input-example"
          allCaps={true}
          labelCopy="Your email"
          data-tid="the-email-input"
          placeholder="example@ethoslife.com"
          formChangeHandler={formChangeHandlerStub}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
