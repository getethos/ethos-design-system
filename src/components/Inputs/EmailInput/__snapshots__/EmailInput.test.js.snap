// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput default rendering 1`] = `
Array [
  <label
    className="Caption Theinhardt Medium500 GrayPrimary AllCaps"
    htmlFor="email-input"
  >
    Your email
  </label>,
  <div
    className="Spacer"
    style={
      Object {
        "height": 8,
      }
    }
  />,
  <input
    className="TextInput"
    data-tid="the-email-input"
    name="email-input"
    onBlur={[Function]}
    onChange={[Function]}
    onPaste={[Function]}
    placeholder="example@ethoslife.com"
    type="email"
    value=""
  />,
  "",
  "",
]
`;
